### PR TITLE
Add Navigation Framework for the Linear Skeleton Journey

### PR DESCRIPTION
**Pull request by BSaunders95**
Required for SFA-591

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/annotation/NextController.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.companieshouse.web.accounts.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (ChrisJBurns) [inline src/main/java/uk/gov/companieshouse/web/accounts/annotation/NextController.java]: JavaDoc for this?

File: src/main/java/uk/gov/companieshouse/web/accounts/annotation/PreviousController.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.companieshouse.web.accounts.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (ChrisJBurns) [inline src/main/java/uk/gov/companieshouse/web/accounts/annotation/PreviousController.java]: JavaDoc for this?

File: src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.exception.MissingAnnotationException;
+
+public class Navigator {
+
+    private Navigator() {}
```
Comment (ChrisJBurns) [inline src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java]: I would add a comment here to why you have added a private contructor

File: src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    // Private constructor to prevent instantiation
+    private Navigator() {}
+
+    public static String getNextControllerRedirect(Class clazz, String... pathVars) throws MissingAnnotationException {
+        Annotation nextControllerAnnotation = AnnotationUtils.findAnnotation(clazz, NextController.class);
+        if (nextControllerAnnotation == null) {
+            throw new MissingAnnotationException("Missing @NextController annotation on class " + clazz.toString());
+        }
+
+        Class nextControllerClass = ((NextController) nextControllerAnnotation).value();
```
Comment (paulforsyth) [inline src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java]: worth checking if the class object is null like you're doing for previous?

Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java]: Good point, addressed below

**Pull request merged**